pkg/controller/argocd/role.go

```diff
@@ -50,20 +49,14 @@ func newRoleWithName(name string, cr *argoprojv1a1.ArgoCD) *v1.Role {
 	return sa
 }
 
-func (r *ReconcileArgoCD) getClusterRole(name string) (*v1.ClusterRole, error) {
-	clusterRoleList := &v1.ClusterRoleList{}
-	err := r.client.List(context.TODO(), clusterRoleList)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, clusterRole := range clusterRoleList.Items {
-		if strings.Contains(clusterRole.Name, name) {
-			return &clusterRole, nil
-		}
+func newClusterRole(name string, cr *argoprojv1a1.ArgoCD) *v1.ClusterRole {
+	return &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
```

Comment:
clusterole.ObjectMeta.Name should be assigned with "cr.Name + "-" + argoComponentName". This will ensure that each argocd instance is created with a different clusterrole.

Reply:
These cluster roles will be the same for every instance of Argo CD. Hence the names are not prefixed with `cr.Name` and the cluster roles are shared. But creating separate cluster roles will allow teams to configure their ArgoCD instance without conflicts.

Reply:
I've pushed a commit to create separate cluster roles for each instance